From d5f7880641c979b986f6c4d7b6502bd7e768a9ea Mon Sep 17 00:00:00 2001
From: Arne Redlich <arne.redlich@googlemail.com>
Date: Sun, 9 Mar 2014 21:55:42 +0100
Subject: [PATCH] nfs_normalize_path: don't strip trailing slash from "/"

Otherwise end up with a null string which is not permitted (RFC 1813, 3.2;
the code checks for it right after the now fixed nullification of "/").

Signed-off-by: Arne Redlich <arne.redlich@googlemail.com>
---
 lib/libnfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/libnfs.c b/lib/libnfs.c
index a237f60..9053ab1 100644
--- a/lib/libnfs.c
+++ b/lib/libnfs.c
@@ -994,7 +994,7 @@ static int nfs_normalize_path(struct nfs_context *nfs, char *path)
 
 	/* /$ -> \0 */
 	len = strlen(path);
-	if (len >= 1) {
+	if (len > 1) {
 		if (path[len - 1] == '/') {
 			path[len - 1] = '\0';
 			len--;
-- 
2.0.0

